Sources/MessagingPush/PushContent.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+#if canImport(UserNotifications)
+import UserNotifications
+
+/**
+ The content of a push notification. Single source of truth for getting and setting properties of a push notification.
+
+ Meant to be used by the SDK to set/get the push content but also designed for SDK user to further modify the push
+ if they wish.
+ */
+public class PushContent {
+    public var title: String {
+        didSet {
+            modifyNotificationContent()
+        }
+    }
+
+    public var body: String {
+        didSet {
+            modifyNotificationContent()
+        }
+    }
+
+    public var deepLink: URL? {
+        didSet {
+            modifyNotificationContent()
+        }
+    }
+
+    public let mutableNotificationContent: UNMutableNotificationContent?
+
+    public static func parse(notificationContent: UNNotificationContent) -> PushContent? {
+        let raw = notificationContent.userInfo
+
+        guard let cio = raw["CIO"] as? [AnyHashable: Any], let cioPush = cio["push"] as? [AnyHashable: Any] else {
```

Comment:
The reason to be so generic is for safe type casting. You can assume that if Customer.io sent the push to the SDK, all of this code will execute correctly so we can do something like:

```swift
(raw["CIO"] as! [String: SomeCodable])["push"].link
```

But if the JSON payload from APN is not the format that we expect, a runtime crash could happen. To avoid that we be extra careful by using `Any` and ` if ... as? ...` to prevent the crashes. 

Is that the question you were asking? 

Oh, and `[AnyHashable: Any]` is what `notificationContent.userInfo` gives you so I continued to use that because it's a safe casting type.

Reply:
right, the approach I'd take if this were done in Golang would've been something like

```go
type CIOPushData struct {
  Link string `json:"link"`
}

cio, ok := raw["CIO"]
if !ok {
  return
}

var pushData CIOPushData
err := json.Unmarshal(&pushData, cio["push"])
if err != nil {
  return false
}

...
```
or something similar that basically checks whether it unmarshals into a well-defined struct and then passes that along, instead of continuing to work directly with arbitrary json